libclamav/yara_arena.c

```diff
@@ -168,7 +168,7 @@ int _yr_arena_make_relocatable(
   YR_RELOC* reloc;
   YR_ARENA_PAGE* page;
 
-  size_t offset;
+  ssize_t offset;
```

Comment:
This also doesn't feel right to me. Offsets usually shouldn't be signed.  I also noted that you missed one `offset = va_arg(offsets, size_t);`.  

Our old version of the libyara headers has the offset as an `int32_t`, which of course feels wrong... so I looked in the newer version of the full yara source and it look they changed it at some point to a `uint32_t`: https://github.com/VirusTotal/yara/blob/master/libyara/include/yara/arena.h#L48-L71

I dug further and found that they changed their `offset` variable in their `arena.c` to a `size_t`: https://github.com/VirusTotal/yara/commit/09027e5350c5f6ce8b011318717508fb0ef2eff2

I think the correct fix here would be to keep it as `size_t` and instead change the while condition to `while (offset != EOL)`.